src/lib/pgp-key.c

```diff
@@ -834,16 +834,39 @@ pgp_key_protect(pgp_key_t *                      key,
                 rnp_key_protection_params_t *    protection,
                 const pgp_passphrase_provider_t *passphrase_provider)
 {
+    char passphrase[MAX_PASSPHRASE_LENGTH] = {0};
+
+    if (!passphrase_provider)
+        return false;
+
+    // ask the provider for a passphrase
+    if (!pgp_request_passphrase(passphrase_provider,
+                                &(pgp_passphrase_ctx_t){.op = PGP_OP_PROTECT, .key = key},
+                                passphrase,
+                                sizeof(passphrase))) {
+        return false;
+    }
+
+    bool ret = pgp_key_protect_passphrase(key, format, protection, passphrase);
+    pgp_forget(passphrase, sizeof(passphrase));
+    return ret;
+}
+
+bool
+pgp_key_protect_passphrase(pgp_key_t *                  key,
```

Comment:
I actually considered having a passphrase parameter (instead of a provider) when I implemented `pgp_key_protect` 
 originally. I introduced the passphrase provider concept to deal with cases where a passphrase *may* be required, and key protection is a case where a passphrase *is* definitively always required, so it might make sense for it to just take a passphrase parameter.